crates/node/builder/src/rpc.rs

```diff
@@ -386,15 +378,15 @@ where
 pub trait EthApiBuilderProvider<N: FullNodeComponents>: BuilderProvider<N> + EthApiTypes {
     /// Returns the eth api builder.
     #[allow(clippy::type_complexity)]
-    fn eth_api_builder() -> Box<dyn Fn(&EthApiBuilderCtx<N>) -> Self + Send>;
+    fn eth_api_builder() -> Box<dyn Fn(&EthApiBuilderCtx<N, Self>) -> Self + Send>;
 }
```

Comment:
I assume this will go away, if we do #10769 ?

Reply:
this trait is very useful to avoid the very long generics that node builder has now, ref https://github.com/paradigmxyz/reth/issues/9543